cg/services/analysis_starter/configurator/implementations/nextflow.py

```diff
@@ -0,0 +1,106 @@
+from abc import abstractmethod
+from pathlib import Path
+
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.apps.lims import LimsAPI
+from cg.constants import Workflow
+from cg.constants.nf_analysis import NextflowFileType
+from cg.services.analysis_starter.configurator.abstract_service import Configurator
+from cg.services.analysis_starter.configurator.file_creators.config_creator import (
+    NextflowConfigFileContentCreator,
+)
+from cg.services.analysis_starter.configurator.file_creators.utils import create_file, get_file_path
+from cg.services.analysis_starter.configurator.models.nextflow import NextflowCaseConfig
+from cg.store.models import Case
+from cg.store.store import Store
+
+
+class NextflowConfigurator(Configurator):
+
+    def __init__(self, config: any, store: Store, housekeeper_api: HousekeeperAPI, lims: LimsAPI):
+        self.root_dir: str = config.root_dir
+        self.store: Store = store
+        self.housekeeper_api: HousekeeperAPI = housekeeper_api
+        self.lims: LimsAPI = lims
+        self.config_content_creator = NextflowConfigFileContentCreator(
+            store=self.store,
+            platform=config.platform,
+            workflow_config_path=config.workflow_config_path,
+            resources=config.resources,
+            account=config.slurm.account,
+        )
+
+    def create_config(self, case_id: str) -> NextflowCaseConfig:
+        """Create a Nextflow case config."""
```

Comment:
in a nutshell this is what all Nextflow pipelines need as configuration. All functions are the same except for `_do_pipeline_specific_actions` (open for new names for this function), which is defined as an abstract method here and will be different in each implementation